include/sys/abd.h

```diff
@@ -135,9 +175,29 @@ abd_zero(abd_t *abd, size_t size)
 /*
  * ABD type check functions
  */
-boolean_t abd_is_linear(abd_t *);
-boolean_t abd_is_gang(abd_t *);
-boolean_t abd_is_linear_page(abd_t *);
+static inline boolean_t
+abd_is_linear(abd_t *abd)
+{
+	return ((abd->abd_flags & ABD_FLAG_LINEAR) != 0);
+}
+
+static inline boolean_t
+abd_is_linear_page(abd_t *abd)
+{
+	return ((abd->abd_flags & ABD_FLAG_LINEAR_PAGE) != 0);
+}
+
+static inline boolean_t
+abd_is_gang(abd_t *abd)
+{
+	return ((abd->abd_flags & ABD_FLAG_GANG) != 0);
+}
+
+static inline uint_t
+abd_get_size(abd_t *abd)
+{
+	return (abd->abd_size);
+}
```

Comment:
I originally created this function because we hid the definition on the ABD struct in abd_impl.h. Technically we don't need this anymore, but it can stay if everyone wants to keep it.

Reply:
Yeah, I could go either way but I figured keeping it as an accessor doesn't hurt.